2022/incoming/santa-is-on-github.pod

```diff
@@ -0,0 +1,132 @@
+Author: Thibault DUPONCHELLE <thibault.duponchelle@gmail.com>
+Title: Santa is on GitHub
+Topic: Pithub
+
+=encoding utf8
+
+=head1 Santa is on GitHub
+
+Last year, Santa lost his mind with missing gifts, hence he gathered a council with the elves and they decided to make the process more "rigorous" in 2023.
+This year, it has been decided to use a GitHub repository to organize and track the preparation of gifts for Christmas.
+
+Santa has created a repository called "gifts" in his account "santa".
+
+Each time an elf has finished preparing a toy, he commits a quick note and a helpful message, e.g. "A cowboy for James".
+
+So far, "What is the relation with Perl?" you might ask. It's just a great opportunity to discover and use L<Pithub|https://metacpan.org/pod/Pithub> :)
+
+=head2 Look at last prepared toys (commits)
+
+Every once a while, Santa -- or the elves themselves -- want to check produced items, so they use Pithub for that:
+
+=begin perl
+
+	use Pithub::Repos ();
+	
+	my $repos  = Pithub::Repos->new;
+	my $result = $repos->commits->list( user => 'santa', repo => 'gifts' );
+	
+	while ( my $row = $result->next ) {
+	    print $row->{commit}->{message} . "\n";
+	}
+
+=end perl
+
+It produces this kind of output:
+
+	A bicycle for Tom
+	A Barbie for Lisa
+	A Transformer for Billy
+
+=head2 Gifts should remain a surprise!
+
+But wait, a public GitHub repository means any little young programmer frequenting GitHub will know in advance his Christmas present?! This is not good at all, so Santa decided to change visibility of the repository to "private".
+
+It seemed at first to work well but, sadly, even though Santa and the elves could still access the repo via the GitHub UI, they were no longer able to list the commits via Pithub.
+
+They started to debug by adding some error handling:
+
+=begin perl
+
+	unless ( $result->success ) {
+	    printf "Error: %s\n", $result->response->status_line;
+	    exit 1;
+	}
+
+=end perl
+
+This confirmed that the repository is "not visible" with `Error: 404 Not Found`
```

Comment:
Are the backticks going to work here? Should it be `C<Error: 404 Not Found>`?

Reply:
Yes it should